cloud/apitoken/apitoken.go

```diff
@@ -0,0 +1,257 @@
+	}
+
+	fmt.Println("\nPlease select the api token:")
+
+	apiTokenMap := map[string]astrocore.ApiToken{}
+	for i := range apiTokens {
+		index := i + 1
+		var role string
+		for _, tokenRole := range apiTokens[i].Roles {
+			if tokenRole.EntityType == "DEPLOYMENT" {
+				role = tokenRole.Role
+			}
+		}
+
+		table.AddRow([]string{
+			strconv.Itoa(index),
+			apiTokens[i].Name,
+			role,
+			apiTokens[i].Description,
+			apiTokens[i].Id,
+			apiTokens[i].CreatedAt.Format(time.RFC3339),
+			apiTokens[i].UpdatedAt.Format(time.RFC3339),
+		}, false)
+
+		apiTokenMap[strconv.Itoa(index)] = apiTokens[i]
+	}
+
+	table.Print(os.Stdout)
+	choice := input.Text("\n> ")
+	selected, ok := apiTokenMap[choice]
+	if !ok {
+		return astrocore.ApiToken{}, ErrInvalidAPITokenKey
+	}
+	return selected, nil
+}
+
+func DeleteDeploymentAPIToken(apiTokenID, deployment string, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+
+	if apiTokenID == "" {
+		apiTokens, err := GetDeploymentAPITokens(client, deployment, apiTokenPagnationLimit)
+		if err != nil {
+			return err
+		}
+		apiToken, err := getAPIToken(apiTokens)
+		if err != nil {
+			return err
+		}
+		apiTokenID = apiToken.Id
+	}
+
+	resp, err := client.DeleteDeploymentApiTokenWithResponse(httpContext.Background(), ctx.Organization, deployment, apiTokenID)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Astro ApiToken %s was successfully deleted from deployment %s\n", apiTokenID, deployment)
+	return nil
+}
+
+// Returns a list of all of a deployments apiTokens
+func GetDeploymentAPITokens(client astrocore.CoreClient, deployment string, limit int) ([]astrocore.ApiToken, error) {
+	offset := 0
+	var apiTokens []astrocore.ApiToken
+
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return nil, err
+	}
+
+	for {
+		resp, err := client.ListDeploymentApiTokensWithResponse(httpContext.Background(), ctx.Organization, deployment, &astrocore.ListDeploymentApiTokensParams{
+			Offset: &offset,
+			Limit:  &limit,
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		apiTokens = append(apiTokens, resp.JSON200.ApiTokens...)
+
+		if resp.JSON200.TotalCount <= offset {
+			break
+		}
+
+		offset += limit
+	}
+
+	return apiTokens, nil
+}
+
+// Prints a list of all of an deployments apiTokens
+//
+//nolint:dupl
+func ListDeploymentAPITokens(out io.Writer, client astrocore.CoreClient, deployment string) error {
+	table := printutil.Table{
+		Padding:        []int{30, 50, 10, 50, 10, 10, 10},
+		DynamicPadding: true,
+		Header:         []string{"NAME", "DESCRIPTION", "ID", "DEPLOYMENT ROLE", "CREATE DATE", "UPDATE DATE"},
+	}
+	apiTokens, err := GetDeploymentAPITokens(client, deployment, apiTokenPagnationLimit)
```

Comment:
if there are no api tokens, we should display an error message, rather than an empty table with only headers